### ISSUE TITLE
add popup when new mesage

### ISSUE THREAD

**Issue by marian790** [nextcloud/talk-desktop#290]
Pleas add popup window also with option to bring to taskbar and play custom sound (like lanmessenger), when new message or call

**Comment by ShGKme** [comment]
Can you describe what you mean by the popup window for a new message?

**Comment by marian790** [comment]
> Can you describe what you mean by the popup window for a new message?

talk app siting in tray (is hidden from user eyes). so, when get user new message, talk app go from tray to taskbar and show some window notification/bring talk from tray to backgroun or foreground over all aplication in desktop and play custom soud (new user option in settings talk app). like lanmessenger or another chat clients. chats are more than video call in SOHO.

**Comment by ShGKme** [comment]
I don't think we should bring talk from the system tray to the taskbar and especially open the Talk window on any message. If a user minimizes or closes the window, it should not break in. And I don't know of any messenger that does this (Teams, Slack, Discord, Telegram, WhatsApp, Viber, Signal).

But I agree we should inform more about the new message:

- Native notification. It works now but has known some problems.
- Fix a badge in the taskbar icon. Sometimes it doesn't work, a known issue.
- Add a badge in the system tray icon.

Speaking about a call notification... Yep, I was thinking about adding a popup window here. It could be useful.

**Comment by ShGKme** [comment]
For a custom sound — also doesn't seem needed to me, and we don't have this option in Web and Mobile clients.

**Comment by marian790** [comment]
I work for SOHO (25 users) and another companies as admin. Long time i search some chat apk with LAN/WAN connection, multiplatform (win, mac, android, ...), on site server without 3rd party org, with active develop status.

And, many SOHO, need most time only chat, between users. And if user is busy (at work he should), he needs a more aggressive and at the same time fully adjustable way of notification, such as a flashing taskbar, a popup with the sent text, a sound notification.  Pleas try test lanmessenger, you will see.

employees often communicate with each other, regarding messages that need to be resolved, for example, when the recipient of the message has a phone call, boss call (not watching the tray), someone can send him a text message, that must be read immediately, or if the recipient of the message is not at the PC at the moment and cannot check the chat every time, when he returns to PC to see, if someone has sent him something.

it is more efficient for new chat messages and alo for call, to stay in the taskbar or as a popup window.


> I don't think we should bring talk from the system tray to the taskbar and especially open the Talk window on any message. If a user minimizes or closes the window, it should not break in. And I don't know of any messenger that does this (Teams, Slack, Discord, Telegram, WhatsApp, Viber, Signal).
> 
> But I agree we should inform more about the new message:
> 
>     * Native notification. It works now but has known some problems.
> 
>     * Fix a badge in the taskbar icon. Sometimes it doesn't work, a known issue.
> 
>     * Add a badge in the system tray icon.
> 
> 
> Speaking about a call notification... Yep, I was thinking about adding a popup window here. It could be useful.

**Comment by marian790** [comment]
TALK application sounds are played as system sounds and this can be a problem, if I need to set a separate volume for the TALK apk sound notifications only in the sound mixer (like ringing tone, popup sound volume).



> For a custom sound — also doesn't seem needed to me, and we don't have this option in Web and Mobile clients.

**Comment by ShGKme** [comment]
> TALK application sounds are played as system sounds and this can be a problem, if I need to set a separate volume for the TALK apk sound notifications only in the sound mixer (like ringing tone, popup sound volume).
> 
> > For a custom sound — also doesn't seem needed to me, and we don't have this option in Web and Mobile clients.

Talk Desktop must have its own notification and call notification sounds, the same as in the web browser... Are you sure, you have only native notification sound? What OS are you using?

**Comment by marian790** [comment]
![not in sound mixer](https://github.com/nextcloud/talk-desktop/assets/60311864/c4e0781f-c20e-4523-953c-1290fc4a4093)

**Comment by marian790** [comment]
ok, TALK use windows message center.... :( also notifications are delayed about 0,5-1 minute....

**Comment by drlight17** [comment]
> And I don't know of any messenger that does this

E.g. Spark (XMPP) does. It is really a good feature for corporate chat app.
P.S. Did it in [my app](https://github.com/drlight17/talk-electron).

**Comment by ShGKme** [comment]
Unfortunately, after discussions, we've decided not to implement this feature.

- From the desktop client, we provide sound notification and OS-level notification, which gives users control over how they want to be notified of new messages
- We also have mobile clients that provide notifications and may even send notifications to wearable devices
- We have a popup window for a call notification., as incoming calls are more time-sensitive and typically require immediate attention
- However, forcibly bringing the app window for regular messages is too disrupting for users, especially if it is a window with input fields

Also, if this feature is implemented, it should be available on all the clients, including iOS and Android mobile clients.

We might reconsider our decision if/when we have more features for "important" messages/conversations.

**Comment by pindrow** [comment]
Right, desktop talk application no have any popup window or buzz. add this feature only in desktop talk application same lan messenger or old yahoo classic window messenger.
